### PR TITLE
CMake: Add CMake support

### PR DESCRIPTION
**Pull request by rajkan01**
- Added CMakeLists.txt file to filesystem example so that it can build via CMake
- Update mbed-os.lib to latest master to pick CMake changes
- Add CMake test and update Mbed CLI build test in `.travis.yml`

@0xc0170 @hugueskamba

### CODE REVIEW

**Review by hugueskamba — COMMENTED** [review]
Shouldn't the existing `.travis.yml` be modified to include CMake testing? Also shouldn't the Mbed OS reference be updated?

**Comment by hugueskamba** [comment]
`.travis.yml` needs to be modified to include CMake tests.

**Comment by rajkan01** [comment]
> 
> 
> `.travis.yml` needs to be modified to include CMake tests.

@hugueskamba I have added CMake test into `.travis.yml`, please re-review

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-language: python
-python: 2.7
-dist: trusty
+language: python sh
```
Comment (hugueskamba) [inline .travis.yml:1]: Please add the copyright notice.

**Review by hugueskamba — APPROVED** [review]

**Comment by rajkan01** [comment]
@evedon Could you review this PR

**Review by evedon — APPROVED** [review]

**Review by 0xc0170 — APPROVED** [review]

**Comment by 0xc0170** [comment]
As approved and CI completed, I'll merge this

**Pull request merged**